### DISCUSSION TITLE
如何在secertflow增加新的PSI协议

### DISCUSSION THREAD

**Discussion by 123abcs** [secretflow/secretflow#1517] [Q&A]
如何在secertflow增加新的PSI协议

**Comment by huocun-ant** [comment]
当前Secretflow的PSI协议都在 https://github.com/secretflow/psi 这里，可以先看看已有的几个协议，
如果要在该仓实现新协议：
1. 建议先可以先实现算法核心，两方的输入是vector\<string\>，结果是输入的对应的索引，即vector\<uint32\>
2. 需要有单测，单测可以比较容易的看出来算法接口的用法和正确性
3. 算法最好有对应的论文方便算法同学review
4. 核心功能实现后可以提merge request给 https://github.com/secretflow/psi
